src/vsr/replica.zig

```diff
@@ -9078,6 +9097,14 @@ pub fn ReplicaType(
             if (self.superblock.staging.vsr_state.sync_op_max == 0) {
                 return true;
             } else {
+                // Trailers/manifest haven't yet been synced.
+                if (self.client_sessions_checkpoint.size_transferred <
+                    self.superblock.working.client_sessions_reference().trailer_size or
+                    !self.grid.free_set.opened or !self.state_machine_opened)
```

Comment:
It seems like this is _just_ `!self.state_machine_opened`? Unopened client sessions should imply unopened state machine I think

Reply:
Yes, just `!self.state_machine_opened` would suffice since these steps happen sequentially! That is precisely what I'd done before but I saw this comment and thought it could be a good idea to separate them out?
https://github.com/tigerbeetle/tigerbeetle/blob/56193862533052b686396f521298920812f75d9d/src/vsr/replica.zig#L829-L835

We could also take this up when we _actually_ start doing these concurrently. I'm okay doing either!

Reply:
Went with just simplifying it down to `!self.state_machine_opened`: https://github.com/tigerbeetle/tigerbeetle/pull/2221/commits/f6fa8466f242f544725e4f8941de44811f1c91e1